### ISSUE TITLE
intersectionPoints not reliable?

### ISSUE THREAD

**Issue by frankrolf** [typemytype/drawbot#239]
A student was asking me about the `BezierPath` method `intersectionPoints`, and I tried to explain it in class. I noticed that the result is not always reliable/expected.

See the following code:
```
fill(None)
stroke(0)
strokeWidth(2)

# this example will work as expected
polygon_1 = BezierPath()
polygon_1.moveTo((100, 700))
polygon_1.lineTo((500, 700))
polygon_1.lineTo((101, 900))  # works
# polygon_1.lineTo((100, 900))  # doesn’t work
polygon_1.lineTo((500, 500))
polygon_1.closePath()

drawPath(polygon_1)
print(polygon_1.intersectionPoints())

# this example won’t work
polygon_2 = BezierPath()
polygon_2.moveTo((400, 300))
polygon_2.lineTo((800, 300))
polygon_2.lineTo((400, 500))
polygon_2.lineTo((800, 100))
polygon_2.closePath()

drawPath(polygon_2)
print(polygon_2.intersectionPoints())
```

![intersection](https://user-images.githubusercontent.com/2049645/48071565-ee2f5680-e1da-11e8-809d-f93a6747a60b.png)

The top polygon will correctly report an intersection.
The bottom polygon will not report an intersection.

The issue seems to be related to coinciding x-values in the 2nd and 4th point.

**Comment by typemytype** [comment]
ow you example fails dramatically with booleanOperations

see https://github.com/typemytype/booleanOperations/issues/48

**Comment by anthrotype** [comment]
Fixed by typemytype/booleanOperations#51
